synapse/cores/common.py

```diff
@@ -884,12 +884,9 @@ def addTufoTag(self, tufo, tag, asof=None):
         rows = s_tags.genTufoRows(tufo,tag,valu=asof)
         if rows:
             self.addRows(rows)
-            tufo[1].update({ p:v for (i,p,v,t) in rows })
-
-            form = tufo[1].get('tufo:form')
-            valu = tufo[1].get(form)
-
-            self.fire('tufo:tag:add', tufo=tufo, tag=tag, asof=asof)
+            for (i,p,v,t) in rows:
+                tufo[1][p] = v
+                self.fire('tufo:tag:add', tufo=tufo, tag=p.rpartition('|')[2], asof=asof)
```

Comment:
Rather than cutting the strings up (that we just created) maybe we should update genTufoRows() to return a list of the added tags as well as the props? ( for perf ).

tags,rows = s_tags.genTufoRows(tufo,tag,valu=asof)

and then the fire() loop can just fire based on those?